source/base/function_parser.cc

```diff
@@ -176,6 +180,34 @@ namespace internal
   {
     return erfc(value);
   }
+
+  // returns a random value in the range [0,1] initializing the generator
+  // with the given seed
+  double mu_rand_seed(double seed)
+  {
+    static Threads::Mutex rand_mutex;
+    Threads::Mutex::ScopedLock lock(rand_mutex);
+
+    static boost::random::uniform_real_distribution<> uniform_distribution(0,1);
+
+    static std::map<double, boost::random::mt19937> rng_map;
+
+    if (rng_map.find(seed) == rng_map.end())
+      rng_map[seed] = boost::random::mt19937(static_cast<unsigned int>(seed));
```

Comment:
Ah, clever solution. Care to add a comment somewhere in this function what the purpose of the `map` is (namely, to create a separate random number generator for every unique choice of random seed)?